concordium-std/src/test_infrastructure.rs

```diff
@@ -1588,6 +1648,36 @@ impl<State: Serial + DeserialWithState<TestStateApi>> TestHost<State> {
         *self.contract_balance.borrow_mut() = amount;
     }
 
+    pub fn set_self_address(&mut self, address: ContractAddress) {
+        if self.missing_contracts.contains(&address) {
+            fail!("The self_address is marked as a missing contract address.")
+        }
+        self.contract_address = Some(address);
+    }
+
+    pub fn setup_account_balance(
+        &mut self,
+        address: AccountAddress,
+        account_balance: AccountBalance,
+    ) {
+        self.query_account_balances.borrow_mut().insert(address, account_balance);
+    }
+
+    pub fn setup_contract_balance(&mut self, address: ContractAddress, balance: Amount) {
```

Comment:
Document.

Including the preconditions about the self address.

I wonder why this returns a unit instead of a Result? Did you experiment and decide this is more ergonomic?

Reply:
Part of the issue will be that since this is not a macro, the line number that will be reported will probably be in this library, so it might be a bit trickier to find.

Reply:
I expect this to be more ergonomic. We did not have to fail here, but it seems like a better user experience and might catch some copy paste mistakes.
Going with Result would mean the user would have to deal with the error and since it is only ment for unit tests they would panic anyway to fail the test.